frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/shared/document_bloc_extension.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:appflowy/plugins/document/application/doc_bloc.dart';
+import 'package:appflowy_backend/protobuf/flowy-user/protobuf.dart';
+
+extension DocumentBlocExtension on DocumentBloc {
+  bool get isLocalMode {
+    final userProfilePB = state.userProfilePB;
+    final type = userProfilePB?.authenticator ?? AuthenticatorPB.Local;
+    return type == AuthenticatorPB.Local;
+  }
+}
```

Comment:
Very interesting. Why not add a getter in the DocumentBloc itself?

Reply:
Deleted the extension and added this function in the DocumentBloc.